numpy/random/mtrand.pyx

```diff
@@ -4456,7 +4457,11 @@ cdef class RandomState:
                     self._shuffle_raw(n, sizeof(np.npy_intp), stride, x_ptr, buf_ptr)
                 else:
                     self._shuffle_raw(n, itemsize, stride, x_ptr, buf_ptr)
-        elif isinstance(x, np.ndarray) and x.ndim and x.size:
+        elif isinstance(x, np.ndarray):
+            if x.size == 0:
```

Comment:
Could this be `x.size < 2`? Also below.

Reply:
Could that change the random stream? I don't think it's particularly worth optimizing for.

Reply:
It shouldn't. If `x.size == 1`, then `n == 1`, so the loop that draws from `random_interval()` will not actually execute because `range(1, n)` is empty. But also not especially worth optimizing for.